contracts/loan_manager/src/contract.rs

```diff
@@ -183,94 +182,55 @@ impl LoanManager {
         Ok(())
     }
 
-    pub fn add_interest(e: Env) {
+    pub fn add_interest(e: &Env, user: Address) {
         const DECIMAL: i128 = 10000000;
-        /*
-        We calculate interest for ledgers_between from a given APY approximation simply by dividing the rate r with ledgers in a year
-        and multiplying it with ledgers_between. This would result in slightly different total yearly interest, e.g. 12% -> 12.7% total.
-        Perfect calculations are impossible in real world time as we must use ledgers as our time and ledger times vary between 5-6s.
-        */
-        // TODO: we must store the init ledger for loans as loans started on different times would pay the same amount of interest on the given time.
+        let Loan {
+            borrower,
+            borrowed_from,
+            collateral_amount,
+            borrowed_amount,
+            collateral_from,
+            #[allow(unused_variables)]
+            health_factor,
+            unpaid_interest,
+            last_accrual,
+        } = Self::get_loan(e, user.clone());
 
-        let current_ledger = e.ledger().sequence();
+        let borrow_pool_client = loan_pool::Client::new(e, &borrowed_from);
+        let collateral_pool_client = loan_pool::Client::new(e, &collateral_from);
 
-        let key: LoansDataKey = LoansDataKey::LastUpdated;
-        let previous_ledger: u32 = e.storage().persistent().get(&key).unwrap_or(current_ledger); // If there is no previous ledger, use current.
+        let token_ticker = borrow_pool_client.get_currency().ticker;
+        let token_collateral_ticker = collateral_pool_client.get_currency().ticker;
```

Comment:
You could use unstructuring here as well, might clean the code a tiny bit. Up to you.